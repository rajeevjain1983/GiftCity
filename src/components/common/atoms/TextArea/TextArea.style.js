// 9fbef606107a605d69c0edbcd8029e5d 
import { css } from 'styled-components';

const textareaStyles = css`
  .TextArea__error {
    display: flex;
    flex-direction: row;
    padding-bottom: ${props => props.theme.spacing.ELEM_SPACING.XXS};
  }
  .warning-icon {
    /* background: transparent url(${getIconPath('circle-alert-fill')}) no-repeat 0 0; */
    background-size: contain;
    border: none;
    height: 14px;
    width: 16px;
    margin-right: 7px;
    flex-shrink: 0;
  }
`;
export default textareaStyles;
