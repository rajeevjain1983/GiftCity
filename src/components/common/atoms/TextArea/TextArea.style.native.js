// 9fbef606107a605d69c0edbcd8029e5d 
import styled from 'styled-components/native';
import { TextInput } from 'react-native';

export const StyledTextInput = styled(TextInput)`
  height: 150px;
  width: 100%;
  border: 1px solid ${props => props.theme.colors.FOOTER.DIVIDER};
`;

export default StyledTextInput;
