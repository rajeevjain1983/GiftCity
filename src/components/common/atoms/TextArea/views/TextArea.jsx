// 9fbef606107a605d69c0edbcd8029e5d 
import React from 'react';
import PropTypes from 'prop-types';
import BodyCopy from '../../BodyCopy';
import withStyles from '../../../hoc/withStyles';
import StyledTextArea from '../TextArea.style';

export const TextArea = ({
  className,
  id,
  ariaLabel,
  placeholder,
  maxLength,
  inputRef,
  meta,
  dataLocator,
  enableSuccessCheck,
  isRequired,
  errorDataLocator,
  showExplicitError,
  input,
  ...others
}) => {
  const elemValue = input.value;
  const { touched, error } = meta;
  const errorMessagea11yLbl = `textarea__error__${input.name}`;
  const isErrored = (touched && error) || showExplicitError;

  return (
    <label htmlFor={id} className={`${className} input-fields-wrapper`}>
      <textarea
        {...others}
        {...input}
        id={id}
        aria-label={ariaLabel}
        className="TextArea__input"
        name={input.name}
        maxLength={maxLength}
        value={elemValue}
        ref={inputRef}
        placeholder=""
        data-locator={dataLocator}
        aria-required={isRequired}
        aria-describedby={errorMessagea11yLbl}
      />
      <BodyCopy className="TextArea__label" fontFamily="secondary" fontSize="fs12">
        {placeholder}
      </BodyCopy>
      <div className="TextArea__error">
        <div className={isErrored ? 'warning-icon' : ''} aria-disabled="true" />
        <BodyCopy
          color="error"
          component="div"
          fontSize="fs12"
          fontFamily="secondary"
          fontWeight="extrabold"
          role="alert"
          aria-live="assertive"
          data-locator={errorDataLocator}
          id={errorMessagea11yLbl}
        >
          {isErrored ? error || showExplicitError : ''}
        </BodyCopy>
      </div>
      {enableSuccessCheck && <div className="success__checkmark" />}
    </label>
  );
};

TextArea.propTypes = {
  className: PropTypes.string.isRequired,
  id: PropTypes.string,
  ariaLabel: PropTypes.string,
  placeholder: PropTypes.string,
  maxLength: PropTypes.number.isRequired,
  inputRef: PropTypes.node.isRequired,
  meta: PropTypes.shape({}),
  dataLocator: PropTypes.string,
  enableSuccessCheck: PropTypes.bool,
  isRequired: PropTypes.bool,
  errorDataLocator: PropTypes.string,
  input: PropTypes.shape({}).isRequired,
  showExplicitError: PropTypes.string,
};

TextArea.defaultProps = {
  id: '',
  ariaLabel: '',
  placeholder: '',
  errorDataLocator: '',
  dataLocator: '',
  meta: {},
  isRequired: false,
  enableSuccessCheck: false,
  showExplicitError: '',
};

export default withStyles(TextArea, StyledTextArea);
