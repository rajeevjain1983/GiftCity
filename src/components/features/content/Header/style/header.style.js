import { css } from "styled-components";

export default css`
 
.container{
  display: flex;
  padding: ${props=>props.theme.spacing.ELEM_SPACING.SM} 0 0 0;
}
.menu-button{
  margin-top:${props=>props.theme.spacing.ELEM_SPACING.XXS};
  padding:${props=>props.theme.spacing.ELEM_SPACING.XXS};
  display:flex;
  align-items:center;
}
.shadow-line{
  height:8px;
  background-color:${props=>props.theme.colors.common.tColor};
}

@media ${props => props.theme.mediaQuery.large} {
  .navigation-container{
  background:${props=>props.theme.bodyBackgroundColor};
}
}
.brand-logo-container{
  display:flex;
  flex-direction:row;
  align-items: center;
  padding-left:${props=>props.theme.spacing.ELEM_SPACING.SM};
}
  
  .brand-logo {
    padding-left:${props=>props.theme.spacing.ELEM_SPACING.XS};
    width: 130px;
    height: 40px;
  }
  .search-bar{
    max-width: 500px;
  }
  .cart{
    width: 60px;
    padding-right:${props=>props.theme.spacing.ELEM_SPACING.XS};
  }
  

  @media ${props => props.theme.mediaQuery.large} {
    .brand-logo {
    width: 190px;
    height: 55px;
  }
  }
`;
