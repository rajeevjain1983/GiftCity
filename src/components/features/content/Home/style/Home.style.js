import { css } from "styled-components";

export default css`
  width: 100%;
  margin: 0px auto;
  padding:0px 12px;
.product-container{
width:100%;
}
  
`;
