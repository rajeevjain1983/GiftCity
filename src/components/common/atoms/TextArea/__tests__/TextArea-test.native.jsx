// 9fbef606107a605d69c0edbcd8029e5d 
import React from 'react';
import { shallow } from 'enzyme';
import TextArea from '../views/TextArea.native';
import { StyledTextInput } from '../TextArea.style.native';

describe('TextArea component', () => {
  let component;
  let onChangeSpy;
  let onBlurSpy;
  beforeEach(() => {
    onChangeSpy = jest.fn();
    onBlurSpy = jest.fn();
    const props = {
      id: 'input',
      name: 'input',
      meta: { touched: '', error: '' },
      enableSuccessCheck: false,
      label: 'input',
      input: {
        value: 'foo',
        onChange: onChangeSpy,
        onBlur: onBlurSpy,
      },
    };

    component = shallow(<TextArea {...props} />);
  });

  it('should renders correctly', () => {
    expect(component).toMatchSnapshot();
  });

  it('should renders correctly in case of error', () => {
    component.setProps({
      meta: { touched: true, error: 'error' },
    });
    expect(component).toMatchSnapshot();
  });

  it('onChangeText should call input.onChange', () => {
    component.find(StyledTextInput).prop('onChangeText')('test');
    expect(onChangeSpy).toHaveBeenCalledWith('test');
  });

  it('onBlur should call input.onBlur', () => {
    component.find(StyledTextInput).prop('onBlur')();
    expect(onBlurSpy).toHaveBeenCalled();
  });
});
