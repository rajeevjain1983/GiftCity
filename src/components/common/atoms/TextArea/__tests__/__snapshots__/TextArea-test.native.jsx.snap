// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextArea component should renders correctly 1`] = `
<View>
  <Styled(Styled(BodyCopy))
    accessibilityText=""
    color="gray.900"
    dataLocator=""
    fontFamily="secondary"
    fontSize="fs12"
    fontWeight="regular"
    margin={null}
    numberOfLines={null}
    spacingStyles="margin-bottom-XS"
    text="input"
    textDecoration={null}
  />
  <Styled(Component)
    autoCapitalize="sentences"
    id="input"
    keyboardType="default"
    multiline={true}
    name="input"
    numberOfLines={4}
    onBlur={[Function]}
    onChangeText={[Function]}
    onEndEditing={[Function]}
    onFocus={[Function]}
    returnKeyType="next"
    value="foo"
  />
</View>
`;

exports[`TextArea component should renders correctly in case of error 1`] = `
<View>
  <Styled(Styled(BodyCopy))
    accessibilityText=""
    color="gray.900"
    dataLocator=""
    fontFamily="secondary"
    fontSize="fs12"
    fontWeight="regular"
    margin={null}
    numberOfLines={null}
    spacingStyles="margin-bottom-XS"
    text="input"
    textDecoration={null}
  />
  <Styled(Component)
    autoCapitalize="sentences"
    id="input"
    keyboardType="default"
    multiline={true}
    name="input"
    numberOfLines={4}
    onBlur={[Function]}
    onChangeText={[Function]}
    onEndEditing={[Function]}
    onFocus={[Function]}
    returnKeyType="next"
    value="foo"
  />
  <Styled(View)>
    <Styled(View)
      spacingStyles="margin-right-XS"
    >
      <Styled(ImageComp)
        alt=""
        cacheType="immutable"
        height="14px"
        host=""
        isFastImage={false}
        onError={[Function]}
        onLoadStart={[Function]}
        onProgress={[Function]}
        priority="normal"
        resizeMode="contain"
        source={
          Object {
            "testUri": "../../../src/assets/images/alert-triangle.png",
          }
        }
        url=""
        width="16px"
      />
    </Styled(View)>
    <Styled(BodyCopy)
      accessibilityText=""
      color="error"
      dataLocator=""
      fontFamily="secondary"
      fontSize="fs12"
      fontWeight="extrabold"
      margin={null}
      numberOfLines={null}
      text="error"
      textDecoration={null}
    />
  </Styled(View)>
</View>
`;
